### PR TITLE
Update LoadBalancerReadyCondition on deletion

### LINKED ISSUE #3870: LoadBalancerReady condition stuck in `Deleting`
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]
1. Create AWSCluster WC
1. Wait for cluster to become ready
1. Delete the cluster
1. Notice that the LoadBalancerReadyCondition never becomes `Deleted`

_NOTE:_ The finalizer is eventually removed, but the condition is stuck in `Deleting`.

**What did you expect to happen:**

The condition to eventually become `Deleted`.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api-provider-aws version: v1.5.2
- Kubernetes version: (use `kubectl version`): 1.23.13
- OS (e.g. from `/etc/os-release`): ubuntu 22.04

CC @fiunchinho

### PR DESCRIPTION
**Pull request by mnitchev**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

The LoadBalancerReadyCondition is sometimes explicitly patched and other times it only updates the awsCluster object. Furthermore the condition won't be patched by the patch helper in the AWSClusterReconciler, because it's not in the `patch.WithOwnedConditions` list. This change puts the condition in the list and also updates the condition reason to `Deleted` when the LB is not found. Without setting the condition in that case, the reconciler will set condition to `Deleting` and it will never go back to `Deleted`.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3870

**Special notes for your reviewer**:

**Checklist**:

- [x] squashed commits
- [ ] includes documentation
- [x] adds unit tests
- [ ] adds or updates e2e tests

CC @fiunchinho

### CODE REVIEW

**Comment by Ankitasw** [comment]
/hold
until [this](https://kubernetes.slack.com/archives/CD6U2V71N/p1669226823932389) discussion is concluded

**Comment by AverageMarcus** [comment]
@Ankitasw what needs to be concluded with the discussion on Slack?

Is there anything that's blocking this PR from moving ahead?

We've been running with these changes included for a couple weeks now (based on the 1.5 branch) and it's been working as expected.

File: controllers/awscluster_controller.go [hunk 1]
```diff
@@ -171,6 +171,7 @@ func (r *AWSClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 			patch.WithOwnedConditions{Conditions: []clusterv1.ConditionType{
 				infrav1.PrincipalCredentialRetrievedCondition,
 				infrav1.PrincipalUsageAllowedCondition,
+				infrav1.LoadBalancerReadyCondition,
```
Comment (Ankitasw) [inline controllers/awscluster_controller.go:174]: Is this change really needed?
Does setting `LoadBalancerReadyCondition` as done in loadbalancer.go would not suffice?

Comment (mnitchev) [reply controllers/awscluster_controller.go:174]: Our observation was that it didn't work without having the condition in that list. My understanding is that the patch helper will ignore conditions not owned by the controller. At least that's the impression I'm left with by the documentation [here](https://github.com/kubernetes-sigs/cluster-api/blob/333341b1f72e9ac2fdb40b02b0f114054c68cd5b/util/patch/options.go#L60-L64).

I'll double check, since it's been a while since I submitted this change.

Comment (Ankitasw) [reply controllers/awscluster_controller.go:174]: cc @richardcase if you have any opinions

**Comment by richardcase** [comment]
This change seems reasonable to me. Lets check the e2e pass and then merge:

/test pull-cluster-api-provider-aws-e2e

**Comment by richardcase** [comment]
Until the e2e pass:
/hold

**Comment by richardcase** [comment]
/approve

@mnitchev @AverageMarcus - feel free to unhold when the e2e passes.

**Comment by AverageMarcus** [comment]
E2E passed

/hold cancel
